public/app/plugins/panel/datagrid/DataGridPanel.tsx

```diff
@@ -40,7 +40,13 @@ import {
 export interface DataGridProps extends PanelProps<PanelOptions> {}
 
 export function DataGridPanel({ options, data, id, fieldConfig, width, height }: DataGridProps) {
-  const [state, dispatch] = useReducer(datagridReducer, initialState);
+  const [state, dispatch] = useReducer(datagridReducer, {
+    ...initialState,
+
+    // TODO -- also check if the user has permissions?
+    // TODO -- respond to changes in the option
+    editingEnabled: options.enableEditing,
```

Comment:
can useReducer respond to changers in the options?  (my redux skillz are zero)

Reply:
yeah, that can be done. It's a cool new option to add, but I would leave it as a future improvement after the release